app/helpers/auto-approval-check.ts

```diff
@@ -0,0 +1,192 @@
+  testQuota: Quota;
+  toolsQuota: Quota;
+  developmentQuota: Quota;
+  productionQuota: Quota;
+}
+
+// check if request contains quota change
+export const checknoQuotaChange = (currentQuota: Quotas, requestedQuota: Quotas): boolean => {
+  let noQuotaChange = true;
+  // @ts-ignore
+  _each(currentQuota, (quota: Quota, envQuota: keyof Quotas) => {
+    if (!noQuotaChange) return false; // Stop looping if a mismatch has already been found
+    // @ts-ignore
+    _each(quota, (resource: string, resourceName: keyof Quota) => {
+      const currentVal = resource;
+      const requestedVal = requestedQuota[envQuota][resourceName];
+      if (extractNumbers(requestedVal)[0] !== extractNumbers(currentVal)[0]) {
+        noQuotaChange = false;
+        return;
+      }
+    });
+  });
+  return noQuotaChange;
+};
+
+// if quota was changed check if it was undngrade quota request
+// TODO replace isQuotaUpgrade at app/emails/_templates/private-cloud/TeamEditRequest.tsx
+export const checkIfQuotaUpgrade = (currentQuota: Quotas, requestedQuota: Quotas) => {
+  let ifQuotaUpgrade = true;
+  // @ts-ignore
+  _each(currentQuota, (quota: Quota, envQuota: keyof Quotas) => {
+    // @ts-ignore
+    _each(quota, (resource: string, resourceName: keyof Quota) => {
+      const currentVal = resource;
+      const requestedVal = requestedQuota[envQuota][resourceName];
+      ifQuotaUpgrade = extractNumbers(requestedVal)[0] > extractNumbers(currentVal)[0];
+    });
+  });
+  return ifQuotaUpgrade;
+};
+
+const resourceOrders = {
+  cpu: defaultCpuOptionsLookup,
+  memory: defaultMemoryOptionsLookup,
+  storage: defaultStorageOptionsLookup,
+};
+
+// Helper function to check resource utilization
+const checkUtilization = async (
+  requestedQuota: Quotas,
+  currentQuota: Quotas,
+  licencePlate: string,
+  cluster: Cluster,
+  namespace: keyof typeof NamespaceNames,
+  resource: ResourceType,
+) => {
+  const podMetricsData = await getPodMetrics(licencePlate, namespace, cluster);
+  if (podMetricsData.length === 0) return false;
+
+  const { totalUsage } = getTotalMetrics(podMetricsData, resource);
+  const mesUnitsCoeff = resource === 'cpu' ? 1000 : 1024 * 1024;
+  const totalLimit = extractNumbers(currentQuota[NamespaceNames[namespace]][resource])[1] * mesUnitsCoeff;
+
+  // Check quota utilization
+  // namespace already use 85% of total limit(i.e: kube_pod_container_resource_limits/namespace-total-limit)(CPU or memory, or storage)
+  if (totalLimit && totalUsage) {
+    const utilizationPercentage = (totalLimit / totalUsage) * 100;
+    if (utilizationPercentage < 86) {
+      return true; // Auto-approval due to utilization percentage
+    }
+    // Check quota usage
+    // namespace has at least 35% of CPU utilization rate(namespace:container_cpu_usage/namespace_cpu:kube_pod_container_resource_requests:sum > 35%, same idea for memory)
+    const requestedUsage = extractNumbers(requestedQuota[NamespaceNames[namespace]][resource])[0] * mesUnitsCoeff;
```

Comment:
should be `currentQuota`'s?

Reply:
doesn't sounds like this to me 
"Context: This metric assesses how well the namespace is utilizing the resources it has requested. For example, if the namespace has requested 1000 CPU units but is only using 340 units, the Utilization Rate would be 34%. This helps determine whether the resources are being efficiently used or are underutilized."